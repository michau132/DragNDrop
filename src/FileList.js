import React from 'react';
import {
  Table, TableCell, TableRow, TableHead, TableBody,
} from '@material-ui/core';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import FileListItem from './FileListItem';

const StyledWrapper = styled.div`
  overflow-x: hidden;
  overflow-y: auto;
  height: 250px;
`;

const StyledTableCell = styled(TableCell)`
  && {
    padding: 4px 24px;
  }
`;

const FileList = ({ list, removeImage }) => (
  <div>
    <Table>
      <TableHead>
        <TableRow>
          <StyledTableCell>Avatar</StyledTableCell>
          <StyledTableCell>File name</StyledTableCell>
          <StyledTableCell>File type</StyledTableCell>
          <StyledTableCell>Size</StyledTableCell>
          <StyledTableCell>Geo position</StyledTableCell>
          <StyledTableCell align="center">Delete</StyledTableCell>
        </TableRow>
      </TableHead>
    </Table>
    <StyledWrapper>
      <Table>
        <TableBody>
          {
            list.map((img, index) => (
              <FileListItem
                key={img.avatar}
                index={index}
                removeImage={removeImage}
                {...img}
              />
            ))
          }
        </TableBody>
      </Table>
    </StyledWrapper>
  </div>
);

FileList.propTypes = {
  list: PropTypes.arrayOf(
    PropTypes.shape({
      avatar: PropTypes.string.isRequired,
      position: PropTypes.shape({
        lng: PropTypes.number.isRequired,
        lat: PropTypes.number.isRequired,
      }).isRequired,
      name: PropTypes.string.isRequired,
      size: PropTypes.number.isRequired,
      type: PropTypes.string.isRequired,
    }).isRequired,
  ).isRequired,
  removeImage: PropTypes.func.isRequired,
};

export default FileList;
