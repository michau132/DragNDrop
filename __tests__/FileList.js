import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FileList from '../src/FileList';

const props = {
  list: [
    {
      avatar: 'someUrl',
      position: {
        lng: 12.12,
        lat: 12.13,
      },
      name: 'fileName.jpg',
      size: 12353,
      type: '.jpg',
    },
    {
      avatar: 'secondUrl',
      position: {
        lng: 13,
        lat: 50,
      },
      name: 'DESC414.jpg',
      size: 8346,
      type: '.jpg',
    },
  ],
  removeImage: jest.fn(),
};

describe('<FileList/> component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<FileList {...props} />);
  });
  test('should be defined', () => {
    expect(wrapper).toBeDefined();
  });
  test('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
