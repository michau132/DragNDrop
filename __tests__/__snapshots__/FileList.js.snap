// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileList/> component should match snapshot 1`] = `
<div>
  <WithStyles(Table)>
    <WithStyles(TableHead)>
      <WithStyles(TableRow)>
        <Styled(WithStyles(TableCell))>
          Avatar
        </Styled(WithStyles(TableCell))>
        <Styled(WithStyles(TableCell))>
          File name
        </Styled(WithStyles(TableCell))>
        <Styled(WithStyles(TableCell))>
          File type
        </Styled(WithStyles(TableCell))>
        <Styled(WithStyles(TableCell))>
          Size
        </Styled(WithStyles(TableCell))>
        <Styled(WithStyles(TableCell))>
          Geo position
        </Styled(WithStyles(TableCell))>
        <Styled(WithStyles(TableCell))
          align="center"
        >
          Delete
        </Styled(WithStyles(TableCell))>
      </WithStyles(TableRow)>
    </WithStyles(TableHead)>
  </WithStyles(Table)>
  <styled.div>
    <WithStyles(Table)>
      <WithStyles(TableBody)>
        <FileListItem
          avatar="someUrl"
          index={0}
          key="someUrl"
          name="fileName.jpg"
          position={
            Object {
              "lat": 12.13,
              "lng": 12.12,
            }
          }
          removeImage={[MockFunction]}
          size={12353}
          type=".jpg"
        />
        <FileListItem
          avatar="secondUrl"
          index={1}
          key="secondUrl"
          name="DESC414.jpg"
          position={
            Object {
              "lat": 50,
              "lng": 13,
            }
          }
          removeImage={[MockFunction]}
          size={8346}
          type=".jpg"
        />
      </WithStyles(TableBody)>
    </WithStyles(Table)>
  </styled.div>
</div>
`;
